src/main/kotlin/demo/controllers/ApplicationController.kt

```diff
@@ -17,13 +18,42 @@ class ApplicationController(@Autowired private val applicationService: Applicati
 
     @CrossOrigin
     @PostMapping("/apply/{offerId}/user")
-    fun applyForAnOffer(@PathVariable offerId: Int, @RequestBody applicationPayload: ApplicationLoggedInPayload) =
-        applicationService.insertApplicationLoggedInUser(offerId, applicationPayload)
+    fun applyForAnOffer(@PathVariable offerId: Int, @RequestBody applicationPayload: ApplicationLoggedInPayload): ResponseEntity<Void> {
```

Comment:
Czemu zwracamy ResponseEntity<Void> zamiast Stringa z wiadomością pokroju udało ci się zaaplikować?

Reply:
ResponseEntity ma od razu kod, więc na froncie nie trzeba będzie się bawić w parsowanie Stringów. Ale fakt, moglibyśmy zwracać ResponseEntity<String>, nie byłoby to wtedy mylące.

Reply:
Dobra to my zwracamy to URI jako voida czy też my nic nie zwracamy a to uri jest jakoś inaczej wyciągane ?

Reply:
Szczerze Ci powiem, że nie mam zielonego pojęcia. Nie wiem też, po co to URI zwracamy, bo to tak trochę takie nie wiem. XD

Reply:
Taka konwencja, że wysyłasz response Created z URI do stworzonego zasobu w nagłówku location, to samo jest w crudzie do ofert chłopaki jakby co, ale to można jeszcze ustalić xd